pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/metrics/AuthenticationMetricsToken.java

```diff
@@ -0,0 +1,97 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.authentication.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import java.util.concurrent.TimeUnit;
+import org.apache.pulsar.common.stats.MetricsUtil;
+
+public class AuthenticationMetricsToken extends AuthenticationMetrics {
+
+    @Deprecated
+    private static final Counter expiredTokenMetrics = Counter.build()
+            .name("pulsar_expired_token_total")
+            .help("Pulsar expired token")
+            .register();
+    public static final String EXPIRED_TOKEN_COUNTER_METRIC_NAME = "pulsar.authentication.token.expired.count";
+    private LongCounter expiredTokensCounter;
+
+    @Deprecated
+    private static final Histogram expiringTokenMinutesMetrics = Histogram.build()
+            .name("pulsar_expiring_token_minutes")
+            .help("The remaining time of expiring token in minutes")
+            .buckets(5, 10, 60, 240,
+                    TimeUnit.DAYS.toMinutes(1),
+                    TimeUnit.DAYS.toMinutes(7),
+                    TimeUnit.DAYS.toMinutes(14),
+                    TimeUnit.DAYS.toMinutes(30),
+                    TimeUnit.DAYS.toMinutes(90),
+                    TimeUnit.DAYS.toMinutes(180),
+                    TimeUnit.DAYS.toMinutes(270),
+                    TimeUnit.DAYS.toMinutes(365))
+            .register();
+    public static final String EXPIRING_TOKEN_HISTOGRAM_METRIC_NAME = "pulsar.authentication.token.expiry.duration";
+    private DoubleHistogram expiringTokenSeconds;
+
+    public AuthenticationMetricsToken(OpenTelemetry openTelemetry, String providerName,
+                                      String authMethod) {
+        super(openTelemetry, providerName, authMethod);
+
+        var meter = openTelemetry.getMeter(AuthenticationMetrics.INSTRUMENTATION_SCOPE_NAME);
+        expiredTokensCounter = meter.counterBuilder(EXPIRED_TOKEN_COUNTER_METRIC_NAME)
+                .setDescription("The total number of expired tokens")
+                .setUnit("{token}")
+                .build();
+        expiringTokenSeconds = meter.histogramBuilder(EXPIRING_TOKEN_HISTOGRAM_METRIC_NAME)
+                .setDescription("The remaining time of expiring token in seconds")
+                .setUnit("s")
+                .build();
+    }
+
+    public void recordTokenDuration(Long durationMs) {
+        if (durationMs == null) {
+            // Special case signals a token without expiry. OpenTelemetry supports reporting infinite values.
+            expiringTokenSeconds.record(Double.POSITIVE_INFINITY);
```

Comment:
nit: curious to know how we display this infinite metric in the dashboard. Is this ok?